### PR TITLE
docs(storage): add documentation for blob and bucket name property

### LINKED ISSUE #82: Documentation: Blob name is not documented as a property
Received an internal request to update Storage Documentation to include Blob.name property.

### PR DESCRIPTION
**Pull request by HemangChothani**
Fixes #82

### CODE REVIEW

**Comment by HemangChothani** [comment]
@googlebot I signed it!

File: google/cloud/storage/blob.py [hunk 1]
```diff
@@ -193,6 +194,24 @@ def __init__(
...
+    @property
+    def name(self):
+        """ Get the blob's name.
+
+        :rtype: str
+        :returns: The name for the blob object.
+        """
+        return self._name
+
+    @name.setter
```
Comment (frankyn) [inline google/cloud/storage/blob.py]: Recommend removing setter.

File: google/cloud/storage/bucket.py [hunk 2]
```diff
@@ -512,10 +512,29 @@ def __init__(self, client, name=None, user_project=None):
...
+    def name(self):
+        """ Get the bucket's name.
+
+        :rtype: str
+        :returns: The name for the bucket object.
+        """
+        return self._name
+
+    @name.setter
+    def name(self, value):
```
Comment (frankyn) [inline google/cloud/storage/bucket.py]: Here as well

File: google/cloud/storage/blob.py (hunk 1 above)
Comment (HemangChothani) [reply google/cloud/storage/blob.py]: @frankyn  If i remove setter then it throws an error like `AttributeError: can't set attribute`

Comment (frankyn) [reply google/cloud/storage/blob.py]: Is this why you recommended only adding pydocs instead? If so, then let's do that instead.

Comment (HemangChothani) [reply google/cloud/storage/blob.py:206]: @frankyn  Yes , that's why i recommend doc static doc string. i will update the PR. Thanks.

File: google/cloud/storage/bucket.py [hunk 3]
```diff
@@ -505,13 +505,18 @@ class Bucket(_PropertyMixin):
...
+            Get the bucket's name.
+        """
         name = _validate_name(name)
         super(Bucket, self).__init__(name=name)
         self._client = client
         self._acl = BucketACL(self)
         self._default_object_acl = DefaultObjectACL(self)
         self._label_removals = set()
         self._user_project = user_project
+        self._name = name
```
Comment (frankyn) [inline google/cloud/storage/bucket.py:519]: is this still necessary?

**Pull request merged**